yt_dlp/extractor/bilibili.py

```diff
@@ -86,18 +100,74 @@ def extract_formats(self, play_info):
-                           f'you have to login or become premium member to download them. {self._login_hint()}')
+        if formats:
+            self.check_missing_formats(play_info, formats)
 
+        fragments = traverse_obj(play_info, ('durl', lambda _, v: url_or_none(v['url']), {
+            'url': ('url', {url_or_none}),
+            'duration': ('length', {lambda x: float_or_none(x, scale=1000)}),
+            'filesize': ('size', {int_or_none}),
+        }))
+        if fragments:
+            formats.append({
+                'url': fragments[0]['url'],
+                'filesize': sum(traverse_obj(fragments, (..., 'filesize'))),
+                **({
+                    'fragments': fragments,
+                    'protocol': 'http_dash_segments'
+                } if len(fragments) > 1 else {}),
+                **traverse_obj(play_info, {
+                    'quality': ('quality', {int_or_none}),
+                    'format_id': ('quality', {str_or_none}),
+                    'format': ('quality', {lambda x: format_names.get(x)}),
+                    'resolution': ('quality', {lambda x: format_names.get(x)}),
+                    'duration': ('timelength', {lambda x: float_or_none(x, scale=1000)}),
+                }),
+            })
         return formats
 
-    def _download_playinfo(self, video_id, cid, headers=None):
+    def _get_wbi_key(self, video_id):
+        if time.time() < self._wbi_key_cache.get('ts', 0) + self._WBI_KEY_CACHE_TIMEOUT:
+            return self._wbi_key_cache['key']
+
+        session_data = self._download_json(
+            'https://api.bilibili.com/x/web-interface/nav', video_id, note='Downloading wbi sign')
+
+        lookup = ''.join(traverse_obj(session_data, (
+            'data', 'wbi_img', ('img_url', 'sub_url'),
+            {lambda x: x.rpartition('/')[2].partition('.')[0]})))
+
+        mixin_key_enc_tab = [
+            46, 47, 18, 2, 53, 8, 23, 32, 15, 50, 10, 31, 58, 3, 45, 35, 27, 43, 5, 49,
+            33, 9, 42, 19, 29, 28, 14, 39, 12, 38, 41, 13, 37, 48, 7, 16, 24, 55, 40,
+            61, 26, 17, 0, 1, 60, 51, 30, 4, 22, 25, 54, 21, 56, 59, 6, 63, 57, 62, 11,
+            36, 20, 34, 44, 52
+        ]
+
+        self._wbi_key_cache.update({
+            'key': ''.join(lookup[i] for i in mixin_key_enc_tab)[:32],
+            'ts': time.time(),
+        })
+        return self._wbi_key_cache['key']
+
+    def _sign_wbi(self, params, video_id):
+        params['wts'] = round(time.time())
+        params = {
+            k: ''.join(filter(lambda char: char not in "!'()*", str(v)))
+            for k, v in sorted(params.items())
+        }
+        query = urllib.parse.urlencode(params)
+        params['w_rid'] = hashlib.md5(f'{query}{self._get_wbi_key(video_id)}'.encode()).hexdigest()
+        return params
+
+    def _download_playinfo(self, bvid, cid, headers=None, qn=None):
+        params = {'bvid': bvid, 'cid': cid, 'fnval': 4048}
```

Comment:
Hey, I want to get to reviewing this soon, but something I just discovered is that passing `'try_look': 1` here (and calling the api always instead of using the js playinfo data) it is possible to get 1080p videos without an account.

Would you be interested in exploring this further and possibly incorporating it into either this or a follow-up PR?

Reply:
I agree it is definitely worth looking at.
Still, I'll leave it to a different PR, since it is not a critical issue that needs to be addressed now, and it may take some time to test and understand the behavior of the API.